src/MyRecipes.js

```diff
@@ -0,0 +1,53 @@
+import React, { Component } from "react";
+import firebase from "firebase";
+import Recipe from "./Recipe";
+
+class MyRecipes extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      myRecipes: [],
+      myRecipesRef: null
+    };
+  }
+
+  // retrieves user's posts for state
+  componentDidMount() {
+    let myRecipesRef = firebase
+      .database()
+      .ref("recipes")
+      .orderByChild("author")
+      .equalTo(firebase.auth().currentUser.displayName);
+    myRecipesRef.on("value", snapshot => {
```

Comment:
Since you are checking against the display name (which is NOT being enforced to be unique), if I sign up with another account with the same user name I can access that other users data! Major security issue.